site_scons/site_tools/compilation_db.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2015 MongoDB Inc.
+#
```

Comment:
Why the extra `site_scons/site_tools` path?

Reply:
scons loads these before sconstruct files, it's annoying to have to have the extra directory but I do not know how to make this work without the extra path

here is some docs on this:
https://scons.org/doc/1.0.1/HTML/scons-user/x3627.html

Reply:
That sucks. I'm pretty territorial about what folders we have in our Git repo, so I really don't want to add something like this as top-level folder.
Isn't there a way to declare a new `Tool` from an imported module (`import compilation_db`)?

Otherwise upstream seems willing to merge this upstream, so that might be the cleanest way forward :)

Reply:
I think the best solution would be to merge with scons directly, I like that idea better and means less code here